Add code skeleton and virtualenv

epelz opened the pull request:
Adds skeleton for the code structure, using https://github.com/RuddockHouse/RuddockWebsite
as a base. Additionally, sets up virtualenv and includes instructions for
virtualenv in `README.md`

Completes:
https://app.asana.com/0/24556964288644/25694911605677
https://app.asana.com/0/24556964288644/25190037185565

epelz committed:
Add code skeleton and virtualenv

Adds skeleton for the code structure, using https://github.com/RuddockHouse/RuddockWebsite
as a base. Additionally, sets up virtualenv and includes instructions for
virtualenv in `README.md`

epelz committed:
Fix commenting in a few places

- Removed newlines after docstrings to follow PEP257.
- Added a missing docstring
- Added TODO comments for the commented database connections.
  Included a hashtag for these -- see https://app.asana.com/0/24556964288644/24556964288705
  for details on this.

allisonkong commented on Donut/templates/layout.html:
```diff
@@ -0,0 +1,31 @@
+  <head>
+    <title>Donut</title>
+    <link rel="stylesheet" type="text/css" 
+          href="{{url_for('static', filename='css/style.css')}}" />
```
Spaces around the `{{` and `}}`?

allisonkong commented on Donut/__init__.py:
```diff
@@ -0,0 +1,36 @@
+from flask import Flask, g
+from sqlalchemy import create_engine
```
Do we want to continue using 

```
from flask import Flask, g, url_for, render_template, ...
from sqlalchemy import text, create_engine, ...
```

Or do we just want to do this:

```
import flask
import sqlalchemy
...

flask.render_template(...)
etc
```

That means we need to prefix things with `flask.`, but we don't need to worry about keeping the import statement updated with whatever functions/variables we're using. At least with flask, we use so many different functions that I think it'll be much easier to just `import flask`. With SQLAlchemy, I'm not sure since we haven't really used ORM before (with the Ruddock website doing `from sqlalchemy import text` is fine since that's literally the only function we ever use).

epelz replied:
Yeah, I'm not really sure. All the flask documentation that I've seen uses `from flask import ...`, which is why we've stuck with it. I kind of like the explicitness of the first approach, though I agree that it becomes a little cumbersome with a lot of functions. Does PEP have a preference?

allisonkong replied:
PEP doesn't really care as long as it's not `from module import *`. It's really up to us, so if `import flask` and `flask.function()` seems cleaner let's go with that.

epelz replied:
Cool, let's try this out then. I'll keep the the blueprint init files (e.g. `Donut/modules/example/__init__.py`) the same, since we'll always only need `flask.Blueprint`.

allisonkong commented on Donut/__init__.py:
```diff
@@ -0,0 +1,36 @@
+
+@app.before_request
+def before_request():
+  """Logic executed before request is processed."""
```
:+1: I looked up the style guide for docstrings and this looks good. I think we should stick with these styles:

```
"""This is a short docstring."""
```

or

```
"""
This is a multiple line
docstring that spans
additional lines.
"""
```

The one I think looks really weird is this (let's not use it):

```
"""This is a
multi line docstring.
"""
```

epelz replied:
Agreed. We'll probably have to change the style somewhat once we add documentation generation, since we'll have to add in more details (e.g. arguments, return). But I'll add the details in once I set that up.

epelz committed:
Small style fix

epelz committed:
Switch style of flask/sqlalchemy imports

With the Ruddock website, we ended up having a ton of flask imports, and it
quickly got unmanageable. So @dkong1796 suggested changing the style of
imports so we don't need to specify all function/variable imports.

Instead of
```
from flask import Flask, g, url_for, render_template, ...
from sqlalchemy import text, create_engine, ...
```

Switch to
```
import flask
import sqlalchemy
...

flask.render_template(...)
etc
```

allisonkong commented:
looks good to me
